### ISSUE TITLE
False positive: Possible method call error when using KernelAbstractions.jl

### ISSUE THREAD

**Issue by harishanand95** [julia-vscode/StaticLint.jl#389]
Hi, when I write this [example](https://juliagpu.github.io/KernelAbstractions.jl/stable/quickstart/#Writing-your-first-kernel) code from KernelAbstractions.jl, I see that kernels have possible method call error in vscode. I think it's confusing the arguments with backend and workgroup size.

```julia
using KernelAbstractions

@kernel function mul2_kernel(A)
    I = @index(Global)
    A[I] = 2 * A[I]
end

dev = CPU()
A = ones(1024, 1024)
ev = mul2_kernel(dev, 64)(A, ndrange=size(A))
synchronize(dev)
all(A .== 2.0)
```
![2024-06-06-1717694769_screenshot_3840x2160](https://github.com/julia-vscode/StaticLint.jl/assets/7456909/8e6cabd2-a557-49ed-8f9a-b38084879761)

Thanks!
